test/unit/storage/validators/QuotaValidator.test.ts

```diff
@@ -0,0 +1,118 @@
+import type { Readable } from 'stream';
+import { PassThrough } from 'stream';
+import type { ValidatorArgs } from '../../../../src/http/auxiliary/Validator';
+import { BasicRepresentation } from '../../../../src/http/representation/BasicRepresentation';
+import { RepresentationMetadata } from '../../../../src/http/representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../../../../src/http/representation/ResourceIdentifier';
+import type { QuotaStrategy } from '../../../../src/storage/quota/QuotaStrategy';
+import type { Size } from '../../../../src/storage/size-reporter/Size';
+import { QuotaValidator } from '../../../../src/storage/validators/QuotaValidator';
+import { guardStream } from '../../../../src/util/GuardedStream';
+import type { Guarded } from '../../../../src/util/GuardedStream';
+import { guardedStreamFrom, readableToString } from '../../../../src/util/StreamUtil';
+
+describe('QuotaValidator', (): void => {
+  let mockedStrategy: QuotaStrategy;
```

Comment:
I'm adding some comments in this file to help with how to have mock typings that can help you, so all these should be combined. Only did it for 1 variable but should be applied for all mocked classes everywhere.

```suggestion
  let mockedStrategy: jest.Mocked<QuotaStrategy>;
```